File: plugins/stream-debugger/src/main/java/com/intellij/debugger/streams/trace/smart/resolve/impl/ResolverFactory.java
package com.intellij.debugger.streams.trace.smart.resolve.impl;

import com.intellij.debugger.streams.trace.smart.resolve.CallTraceResolver;
import org.jetbrains.annotations.NotNull;

/**
 * @author Vitaliy.Bibaev
 */
public class ResolverFactory {
  private ResolverFactory() {
  }

  private static class Holder {
    private static final ResolverFactory INSTANCE = new ResolverFactory();
  }

  public static ResolverFactory getInstance() {
    return Holder.INSTANCE;
  }

  public CallTraceResolver getResolver(@NotNull String callName) {
    switch (callName) {
      case "distinct":
        return new DistinctCallTraceResolver();
      default:
        return new SimplePeekCallTraceResolver();
    }
  }
}

File: plugins/stream-debugger-core/src/com/intellij/debugger/streams/core/lib/ResolverFactory.kt
// Copyright 2000-2017 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package com.intellij.debugger.streams.core.lib

import com.intellij.debugger.streams.core.resolve.ValuesOrderResolver
import com.intellij.debugger.streams.core.wrapper.StreamCallType
import com.intellij.openapi.util.NlsSafe

/**
 * @author Vitaliy.Bibaev
 */
interface ResolverFactory {
  fun getResolver(@NlsSafe callName: String, callType: StreamCallType): ValuesOrderResolver
}